src/module_3/Module3_Milestone1.md

```diff
@@ -0,0 +1,1825 @@
+
+
+```python
+def plot_curves(y_true, y_pred, curve_type="both", dataset_type="train"):
+    
+    fig, axes = plt.subplots(1, 2, figsize=(14, 6))
+    fig.suptitle(f"{dataset_type.capitalize()} Dataset - Performance Curves")
+
+    if curve_type in ["precision-recall", "both"]:
+        # Compute the precision-recall curve and its AUC
+        precision, recall, _ = precision_recall_curve(y_true, y_pred)
+        pr_auc = auc(recall, precision)
+        
+        axes[0].step(recall, precision)
+        axes[0].set_xlabel('Recall')
+        axes[0].set_ylabel('Precision')
+        axes[0].set_title(f'Precision-Recall Curve (AUC={pr_auc:.2f})')
+
+    if curve_type in ["roc", "both"]:
+        # Compute the ROC curve and its AUC
+        fpr, tpr, _ = roc_curve(y_true, y_pred)
+        roc_auc = auc(fpr, tpr)
+
+        axes[1].plot(fpr, tpr)
+        axes[1].set_xlabel('False Positive Rate')
+        axes[1].set_ylabel('True Positive Rate')
+        axes[1].set_title(f'ROC Curve (AUC={roc_auc:.2f})')
+
+    plt.tight_layout()
+    plt.show()
+
+```
+
+
+```python
+y_pred = val_data['global_popularity']
+y_true = val_data['outcome']
+
+
+plot_curves(y_true, y_pred, curve_type="both", dataset_type="train")
+
+```
+
+
+    
+![png](Module3_Milestone1_files/Module3_Milestone1_30_0.png)
+    
+
+
+### 3. Training an initial model with only numerical variables
+I will be trainning a logistic regression model as the target variable is binary.
+Additionally, I start by training some initial models that do not need preprocessing, that is, with only numerical (and binary) variables. 
+
+
+```python
+train_cols = numerical_cols + binary_cols
+```
+
+#### 3.1 Ridge Regression
+
+**Important:** Before training the Ridge Regression model, it is crucial to standardize the data, ensuring each column has a mean of 0 and a standard deviation of 1. This standardization makes all features comparable, giving equal importance to each one in the model. This practice prevents variables with larger scales from disproportionately influencing the model's predictions due to their scale rather than their relevance. Additionally, standardizing the data helps in achieving faster and more stable convergence of the regression algorithm.
+
+
+```python
+# First, standardize the data
+scaler = StandardScaler()
+
+# Fit the scaler on the training data and transform it
+X_train_scaled = scaler.fit_transform(X_train[train_cols])
+
+
+# Apply the same transformation to the validation data
+X_val_scaled = scaler.transform(X_val[train_cols])
+
+```
+
+
+```python
+def plot_curves_with_hyperparameter(ax, y_true, y_pred, label, curve_type="both"):
+    if curve_type in ["precision-recall", "both"]:
+        # Compute the precision-recall curve and its AUC
+        precision, recall, _ = precision_recall_curve(y_true, y_pred)
```

Comment:
You could have change the plot_curves method to accept axes (and create them if not provided). By doing that you only maintain one method